variables.tf

```diff
@@ -1,52 +1,88 @@
+  description = "Name of bucket to hold tf state"
+  type        = string
 }
 
-variable "bucket" {
-  default     = ""
-  description = "Name of bucket to create (do not provide if using `remote_bucket`)"
+variable "dynamo_locktable_name" {
+  default     = "tf-locktable"
+  description = "Name of lock table for terraform"
   type        = string
 }
+
 variable "kms_alias_name" {
-  default     = ""
+  default     = null
   description = "Name of KMS Alias"
   type        = string
 }
 
 variable "kms_key_id" {
-  default     = ""
-  description = "ARN for KMS key for all encryption operations."
+  default     = null
+  description = "ARN for KMS key for all encryption operations (a key will be created if this is not provided)"
   type        = string
 }
 
+variable "lifecycle_rules" {
+  default     = null
+  description = "lifecycle rules to apply to the bucket (set to null to skip lifecycle rules)"
+
+  type = list(object(
+    {
+      id                            = string
+      enabled                       = bool
+      prefix                        = string
+      expiration                    = number
+      noncurrent_version_expiration = number
+  }))
+}
+
 variable "logging_target_bucket" {
   default     = null
-  description = "The name of the bucket that will receive the log objects"
+  description = "The name of the bucket that will receive the log objects (logging will be disabled if null)"
   type        = string
 }
 
 variable "logging_target_prefix" {
-  default     = "TFStateLogs/"
+  default     = null
   description = "A key prefix for log objects"
   type        = string
 }
 
-variable "remote_bucket" {
-  default     = ""
-  description = "If specified, the remote bucket will be used for the backend. A new bucket will not be created"
+variable "tags" {
+  default     = {}
+  description = "Mapping of any extra tags you want added to resources"
+  type        = map(string)
+}
+
+########################################
+# Assume Role Template Vars
+########################################
+variable "create_assumerole_template" {
+  default     = false
+  description = "If true, create a CloudFormation template that can be run against accounts to create an assumable role"
+  type        = bool
+}
+
+
+variable "assumerole_role_name" {
+  default     = "Terraform"
+  description = "Name of role to create in assumerole template"
   type        = string
 }
 
-variable "table" {
-  default     = "tf-locktable"
-  description = "Name of Dynamo Table to create"
+variable "assumerole_role_external_id" {
+  default     = null
+  description = "External ID to attach to role (this is required, a random ID will be generated if not specified here)"
   type        = string
 }
 
-variable "tags" {
-  default     = {}
-  description = "Mapping of any extra tags you want added to resources"
-  type        = map(string)
+variable "assumerole_role_attach_policies" {
+  default     = ["arn:aws:iam::aws:policy/AdministratorAccess"]
```

Comment:
This role is just to access the state, right? This seems excessive

Reply:
No, this is the assumed role in the sub account and Terraform effectively requires AdministratorAccess to do things. You can create a custom role that only allows services you expect to manage with Terraform, but that sounds like a hot mess to me.